standard/game.lua

```diff
@@ -0,0 +1,169 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Game
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local FnUtil = require('Module:FnUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local Info = Lua.import('Module:Info', {requireDevIfEnabled = true})
+
+local GamesData = Info.games
+
+local ICON_STRING = '[[File:${icon}|link=${link}|class=${class}|${size}]]'
+local DEFAULT_SIZE = '25x25px'
+local ICON_PLACEHOLDER = 'LeaguesPlaceholder.png'
+
+local Game = {}
+
+Game.getIdentifierByAbbreviation = FnUtil.memoize(function()
+	return Table.map(GamesData, function(gameIdentifier, gameData)
+		return gameData.abbreviation:lower(), gameIdentifier end)
+end)
+
+Game.getIdentifierByName = FnUtil.memoize(function()
+	return Table.map(GamesData, function(gameIdentifier, gameData) return gameData.name:lower(), gameIdentifier end)
+end)
+
+---Get the identifier of a entered game. Checks against identifiers, full names and abbreviations.
+---@param args {game: string?, useDefault: boolean?}
+---@return string?
+function Game.toIdentifier(args)
+	args = args or {}
+	local gameInput = args.game
+
+	if String.isEmpty(gameInput) and Logic.nilOr(Logic.readBoolOrNil(args.useDefault), true) then
+		return Info.defaultGame
+	elseif String.isEmpty(gameInput) then
+		return
+	end
+
+	gameInput = gameInput:lower()
+
+	return Game.getIdentifierByAbbreviation()[gameInput]
+		or Game.getIdentifierByName()[gameInput]
+end
+
+---Check if a given game is a valid game.
+---@param game: string?
+---@return string?
+function Game.isValid(game)
+	return String.isEmpty(game) or (Game.toIdentifier{game = game} ~= nil)
+end
```

Comment:
Is this correct?
I'm guessing the return should be boolean
should it be an `and` instead of `or` there?

Reply:
if it is empty return true
if it is not empty actually do the check
the idea is to not return false for empty entry

Reply:
but looking at `toIdentifier` we can drop the empty check
```suggestion
---Check if a given game is a valid game.
---@param game: string?
---@return string?
function Game.isValid(game)
	return Game.toIdentifier{game = game} ~= nil
end
```